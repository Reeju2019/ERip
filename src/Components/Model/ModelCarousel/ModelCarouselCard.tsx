import * as React from 'react'

interface IModelCarouselCardProps {
  data: {
    model: string
    image: string
  }
}

const ModelCarouselCard: React.FunctionComponent<IModelCarouselCardProps> = ({ data }) => {
  return (
    <>
      <div className='d-flex flex-column m-2'>
        <div className='p-2 bg-white'>
          <div className='service-img'>
            <img
              src={data.image}
              className='images d-flex justify-content-center'
              alt={data.model}
            />
          </div>
        </div>
        <p className='service-para'>{data.model}</p>
      </div>
    </>
  )
}

export default ModelCarouselCard
