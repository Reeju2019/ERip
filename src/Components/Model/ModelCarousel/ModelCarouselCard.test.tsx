import { render } from '@testing-library/react'
import renderer from 'react-test-renderer'
import ReactDOM from 'react-dom'
import ModelCarouselCard from './ModelCarouselCard'

interface Data {
    model: string
    image: string
}
const data: Data = {
    model: "",
    image: ""
}
test('Model carousel', () => {
    render(<ModelCarouselCard data={data} />)
})
describe('Model carousel rendering specification', () => {
    it('SnapShot testing', () => {
        const component = renderer.create(<ModelCarouselCard data={data} />)
        const tree = component.toJSON()
        expect(tree).toMatchSnapshot()
    })
    it('render without crashing Model carousel', () => {
        const div = document.createElement('div')
        render(<ModelCarouselCard data={data} />)
        ReactDOM.unmountComponentAtNode(div)
    })
})
