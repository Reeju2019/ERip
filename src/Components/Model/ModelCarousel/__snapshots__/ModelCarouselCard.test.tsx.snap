// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Model carousel rendering specification SnapShot testing 1`] = `
<div
  className="d-flex flex-column m-2"
>
  <div
    className="p-2 bg-white"
  >
    <div
      className="service-img"
    >
      <img
        alt=""
        className="images d-flex justify-content-center"
        src=""
      />
    </div>
  </div>
  <p
    className="service-para"
  />
</div>
`;
